Branch-SDK/src/io/branch/referral/ShareLinkManager.java

```diff
@@ -319,31 +322,31 @@ public boolean isEnabled(int position) {
      * Class for sharing item view to be displayed in the list with Application icon and Name.
      */
     private class ShareItemView extends TextView {
-        Context context_;
+        WeakReference<Context> context_;
         final int padding = 5;
         final int leftMargin = 100;
 
         public ShareItemView(Context context) {
             super(context);
-            context_ = context;
+            context_ = new WeakReference<>(context);
             this.setPadding(leftMargin, padding, padding, padding);
             this.setGravity(Gravity.START | Gravity.CENTER_VERTICAL);
-            this.setMinWidth(context_.getResources().getDisplayMetrics().widthPixels);
+            this.setMinWidth(context_.get().getResources().getDisplayMetrics().widthPixels);
         }
 
         public void setLabel(String appName, Drawable appIcon, boolean isEnabled) {
             this.setText("\t" + appName);
             this.setTag(appName);
             if (appIcon == null) {
-                this.setTextAppearance(context_, android.R.style.TextAppearance_Large);
+                this.setTextAppearance(context_.get(), android.R.style.TextAppearance_Large);
```

Comment:
i think ShareLinkBuilder had a memory leak before, so it's better to account for this especially if we're making Activity reachable by weak reference in Branch.java anyway